stl/inc/execution

```diff
@@ -4342,7 +4350,8 @@ struct _No_init_tag {
 }; // tag to indicate that no initial value is to be used
 
 template <class _FwdIt1, class _FwdIt2, class _BinOp, class _Ty>
-_FwdIt2 _Exclusive_scan_per_chunk(_FwdIt1 _First, const _FwdIt1 _Last, _FwdIt2 _Dest, _BinOp _Reduce_op, _Ty& _Val) {
+_FwdIt2 _Exclusive_scan_per_chunk(_FwdIt1 _First, const _FwdIt1 _Last, _FwdIt2 _Dest, _BinOp _Reduce_op, _Ty& _Val,
+    _STD vector<_Ty>& _Intermediate_result) {

```

Comment:
It seems that there should be compiler-time branches. When the intermediate results can be a part of the resulted sequence, perphaps the original logic should be kept.